### DISCUSSION TITLE
Accept `yarn` instead of `npm`

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/sdk#2629] [Feature Requests]
Currently `npm` is hard-coded in `dfx`:
```
$ git grep '"npm"'
src/dfx/src/commands/new.rs:    std::process::Command::new("npm")
src/dfx/src/commands/new.rs:    std::process::Command::new("npm")
src/dfx/src/lib/builders/assets.rs:        let mut cmd = std::process::Command::new("npm");
```

I am asking for support for the (more-or-less) drop-in replacement `yarn`. I had some nice success using it in connection with `vuetify`:
- `git clone https://github.com/nop33/dfinity-vue -b vuetify`
- `dfx start --background --clean`
- `yarn add webpack-cli`
- `env NPM=yarn dfx deploy` (suggested way of using  `yarn` instead of `npm`)
- `open http://localhost:8000/?canisterId=ryjl3-tyaaa-aaaaa-aaaba-cai`

The `assets.rs` invocation works out of the box (this is what the above commands trigger). The `new.rs` would need some adaptation (maybe probing).

An alternative would be to introduce a `--toolchain` argument (in-lieu-of `NPM` env-var): `dfx --toolchain=yarn` would just select the `yarn` incarnation of the commands and could potentially also do the `yarn add webpack-cli` step.

I am happy to send a PR if you think this is a good idea.

**Comment by viviveevee** [comment]
Also referenced in #1584
